### PR TITLE
[Federation] When Deleting a User, Remove Them From Remote Conversations

### PR DESCRIPTION
**Pull request by mdimjasevic**
This makes sure to remove a user from their remote conversation when the user's account is being deleted.

## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] **changelog.d** contains the following bits of information:
   - [x] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.

### CODE REVIEW

File: services/galley/src/Galley/API/Internal.hs
Comment (pcapriotti) [inline services/galley/src/Galley/API/Internal.hs]: Why would that be better? `removeMember` would query the whole conversation all over again, whereas here we already have a conversation value.

File: services/galley/src/Galley/API/Internal.hs
Comment (pcapriotti) [inline services/galley/src/Galley/API/Internal.hs:445]: Don't you need to loop through the pages here too?

**Review by pcapriotti — APPROVED** [review]
Looks good. I left a question, because I'm not sure about how the part about removing the user from teams works. Feel free to merge if there is no problem there.

File: services/galley/src/Galley/API/Internal.hs
Comment (mdimjasevic) [reply services/galley/src/Galley/API/Internal.hs]: To me this looks like duplicating the logic from `removeMember`, and it feels like checks are performed only partially. But I could be wrong.

File: services/galley/src/Galley/API/Internal.hs
Comment (mdimjasevic) [reply services/galley/src/Galley/API/Internal.hs:445]: I left the team deletion code the same (modulo replacing `rcast n` with `n`, where `n` is slightly differently defined, but should have the same value.

Perhaps it would be worth paging through teams as well. My hunch is that whoever wrote this code, assumed a user wouldn't be in more than 100 teams, hence there shouldn't be the next page to process.

File: services/galley/src/Galley/API/Internal.hs
Comment (mdimjasevic) [reply services/galley/src/Galley/API/Internal.hs]: I'll drop this to-do item and leave the code as is; this is slightly out of scope of this PR.

File: services/galley/src/Galley/API/Internal.hs
Comment (mdimjasevic) [inline services/galley/src/Galley/API/Internal.hs:467]: @pcapriotti or @jschaul , can you take a look at this? I changed this after getting a PR approval, because the underlying paging things changed on `develop` in the meantime. I am not sure if I'm doing the paging state threading properly, let alone if there's a nicer way to do it.

**Pull request merged**
